<a href="https://colab.research.google.com/github/ftycity23/DSRP-Research-Project/blob/main/DSRP_Research.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

**File upload of the data** - Emmy Awards Data


**Description of data**: This dataset describes the evolution of the television industry by highlighting key trends and changes over time. TidyTuesday released it as part of an open data project in September 2021. You can also extract the data directly from the Emmy website, where it spans a longer time from 1957 to 2021. The cleaned version of the dataset consists of approximately 30,000 rows and 7 columns, with each row representing one nomination or win, depending on the “type” indicated. The variables included are category, production, type, title, distributor, producer, and year.



In [33]:
!git clone https://github.com/ftycity23/DSRP-Research-Project.git


import pandas as pd
df = pd.read_csv("DSRP-Research-Project/emmys_original.csv")
df.head()

fatal: destination path 'DSRP-Research-Project' already exists and is not an empty directory.


,category,logo,production,type,title,distributor,producer,year,page,page_id
0,Outstanding Character Voice-Over Performance -...,https://www.emmys.com/sites/default/files/styl...,NaN,Nominee,black-ish: Election Special (Part 2),ABC,ABC,2021,0,1
1,Outstanding Character Voice-Over Performance -...,https://www.emmys.com/sites/default/files/styl...,NaN,Nominee,Bridgerton,Netflix,A Netflix Original Series in association with ...,2021,0,2
2,Outstanding Character Voice-Over Performance -...,https://www.emmys.com/sites/default/files/styl...,NaN,Nominee,Family Guy,FOX,20th Television Animation,2021,0,3
3,Outstanding Character Voice-Over Performance -...,https://www.emmys.com/sites/default/files/styl...,NaN,Nominee,Archer,FX Networks,FX Productions,2021,0,4
4,Outstanding Production Design For A Narrative ...,https://www.emmys.com/sites/default/files/styl...,"Elisabeth Williams, Production Designer",Nominee,The Handmaid's Tale,Hulu,"Hulu, MGM, Daniel Wilson Productions, The Litt...",2021,0,5


Importing pandas & matplotlib for set up

In [34]:
import pandas as pd
import matplotlib.pyplot as plt

# Exploring the data itself with pandas by creating a DataFrame

Exploring the average year, median year, and how many Nominees vs. Winners

In [35]:
df['type'].value_counts()

,count
type,
Nominee,23739
Winner,5939


In [36]:
df['year'].mean()

np.float64(2013.6110250016848)

Figuring out the actual year

In [37]:
avgactual_year = int(df['year'].mean())

In [38]:
print(f"The average actual year is {avgactual_year}")

The average actual year is 2013


In [39]:
df['year'].median()

2015.0

# Analyzing the data and inspecting info

Looking at head and info of the data

In [40]:
df.head()

df.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 29678 entries, 0 to 29677
Data columns (total 10 columns):
 #   Column       Non-Null Count  Dtype 
---  ------       --------------  ----- 
 0   category     29678 non-null  object
 1   logo         29678 non-null  object
 2   production   26825 non-null  object
 3   type         29678 non-null  object
 4   title        29678 non-null  object
 5   distributor  29675 non-null  object
 6   producer     29678 non-null  object
 7   year         29678 non-null  int64 
 8   page         29678 non-null  int64 
 9   page_id      29678 non-null  int64 
dtypes: int64(3), object(7)
memory usage: 2.3+ MB


In [41]:
df.isna().sum()

,0
category,0
logo,0
production,2853
type,0
title,0
distributor,3
producer,0
year,0
page,0
page_id,0


# Capturing the number of rows, columns, and the last 5 rows of the dataset.


`df.tail() and df.shape`




In [42]:
df.tail()

,category,logo,production,type,title,distributor,producer,year,page,page_id
29673,BEST CONTINUING PERFORMANCE BY AN ACTRESS - 1957,/sites/all/themes/emmys/img/winners_outsanding...,NaN,Nominee,Jane Wyman Theatre,NBC,NBC,1957,2281,6
29674,BEST CONTINUING PERFORMANCE BY AN ACTRESS - 1957,https://www.emmys.com/sites/default/files/styl...,NaN,Winner,The Loretta Young Show,NBC,NBC,1957,2281,7
29675,BEST CONTINUING PERFORMANCE BY AN ACTRESS - 1957,/sites/all/themes/emmys/img/winners_outsanding...,NaN,Nominee,Lassie,CBS,CBS,1957,2281,8
29676,BEST CONTINUING PERFORMANCE BY AN ACTRESS - 1957,/sites/all/themes/emmys/img/winners_outsanding...,NaN,Nominee,Four Star Playhouse,CBS,CBS,1957,2281,9
29677,BEST CONTINUING PERFORMANCE BY AN ACTRESS - 1957,/sites/all/themes/emmys/img/winners_outsanding...,NaN,Nominee,Mama,CBS,CBS,1957,2281,10


In [43]:
df.shape

(29678, 10)

# Extracting the columns values

Putting them into the variable cols

In [51]:
cols = ['category','production','type','title','distributor','producer','year']

# Checking for missing values within the dataset (insull)

Using
```isnull().sum()```
to figure and count how many rows are missing a specific value within a column



In [44]:
df.isnull().sum()

,0
category,0
logo,0
production,2853
type,0
title,0
distributor,3
producer,0
year,0
page,0
page_id,0


**Missing the 2853 production columns and 3 distributor columns!!!**

# Cleaning Data

## Checking for duplicates and inconsistency in data

## Counting the rows of exact copies from an earlier row

In [50]:
df.duplicated().sum()

np.int64(0)

## Cleaning and dropping the rows that are missing

Creating a df_clean variable to pull all the data from the function in here

In [45]:
df_clean = df.drop(columns=['logo', 'page', 'page_id'])

**Counting the current size of the clean data after dropping the columns initially.**

> Add blockquote



In [56]:
df_clean.shape

(29675, 7)

In [ ]:
df_clean['category'].nunique()

How many unique category values appear

In [46]:
df['category'].value_counts(dropna=False)

,count
category,
"Outstanding Technical Direction, Camerawork, Video For A Miniseries, Movie Or A Special - 2007",117
"Outstanding Technical Direction, Camerawork, Video For A Miniseries, Movie Or A Special - 2008",111
Outstanding Drama Series - 2021,109
Outstanding Animated Program - 2019,102
"Outstanding Technical Direction, Camerawork, Video Control For A Miniseries, Movie Or A Special - 2013",101
...,...
Television Academy Honors - 2012,1
Outstanding Lead Actress In A Drama Series - 2010,1
"Outstanding Costumes For A Variety, Music Program Or A Special - 2009",1


Truncating the years at the end of each category value to just include the name

In [55]:
df['category'] = df['category'].str.replace(r'\s*-\s*\d{4}$', '', regex=True)

## Removing the rows where the distributor comn as now value/ empty value and saves the result within df_clean

In [54]:
df_clean = df_clean.dropna(subset=['distributor'])

Checking the rows that are identical to other rows from the variable 'cols'

In [52]:
df_clean.duplicated(subset=cols).sum()

np.int64(321)